### ISSUE TITLE
Can't add non-VCS packages

### ISSUE THREAD

**Issue by kaedys** [kardianos/govendor#164]
Ran into a very odd roadblock.  I'd extracted a set of packages from one of my apps into a separate project for eventual upload as its own vendorable repository.  However, when trying to vendor the package in the old app (so I could test its functionality, and vendoring was necessary because it imported a package that was also vendored in that project, so it had to be vendored to use the "same" package due to import paths).  The only error I got back was:

`Error: exit status 128`

Not very edifying (issue 1).  So I did what most people do and started putting Printf's everywhere in Govendor until I finally burrowed down to where the issue is, which is vcs.go:61 and, I'm assuming, git.go in Find() somewhere.  Now, this package _did_ have a `.git` folder, because I'd run `git init`, but I hadn't made a commit to the directory yet.  This apparently was something that Govendor's VCS checks aren't accounting for (issue 2).

I suspect this is failing on git.go:42, which tries to run `git show --pretty=format%H@%ai -s`.  When this is run on a git repository that has been initialized (has a `.git` directory) but does _not_ yet have any commits, it returns the error `fatal: your current branch 'master' does not have any commits yet` and exit status 128, and for no readily apparent reason, the `exec` package is returning an error specifying only the exit state and not the actual error message `git` sent to `stdout`.

So, to summarize, the issues:
* If you try to vendor a package that has had `git init` run on it, but has no commits yet, Govendor bombs out.
* When this happens, the returned error message has literally no information at all that could be used to troubleshoot or even identify the problem.

**Comment by kaey** [comment]
Did you try to vendor your project as a tree?
```
govendor add myrepo/mypkg/^
```

Empty directory is not a go package. Error message could be improved though.

**Comment by kaedys** [comment]
The directory wasn't actually empty, it had valid go files in it, I just hadn't made a commit yet.  I didn't try vendoring by tree because the directory in question had no subdirectories.

**Comment by kaedys** [comment]
Dug into this a bit more, as I ran into it again.  It looks like the error received from exec.Command.Run() is of type exec.ExitError.  The `Error()` method of that type returns the output from the `String()` method of the ProcessState field, which is of type `*os.ProcessState`.  This ProcessState does not include any of the actual Stderr output from the process.  

Due to this, I believe a change that would at least fix the cryptic error message would be to change `git.go:42-45` from:
```
	output, err := cmd.CombinedOutput()
	if err != nil {
		return nil, err
	}
```
to:
```
	output, err := cmd.CombinedOutput()
	if err != nil {
		return nil, fmt.Errorf("%s: %s", err, string(output))
	}
```
As that `output` byte slice holds the combined output of Stdout and Stderr.

This wouldn't fix the underlying issue (you can't vendor a library that has been initialized with `git` but doesn't have any commits on it), but would at least make the error clearer.
